### PR TITLE
Add publication approved overlay notification.

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

- #8843

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+				<Link
+					className="mdc-button mdc-button--raised mdc-ripple-upgraded"
+					external
+					href={ serviceURL }
+					inverse
+					linkButton
+				>
+					{ __( 'Enable features', 'google-site-kit' ) }
+				</Link>
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.js:137]: Using the `Button` component here would be wiser IMO, as this is technically and semantically a button. The `Button` component does support links.

```suggestion
				<Button
					href={ serviceURL }
					trailingIcon={ <ExternalIcon width={ 10 } height={ 10 } /> }
					target="_blank"
				>
					{ __( 'Enable features', 'google-site-kit' ) }
				</Button>
```

File: assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.test.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import { Provider as ViewContextProvider } from '../../../components/Root/ViewContextContext';
+import { UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION } from '../datastore/constants';
+import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+
+describe( 'PublicationApprovedOverlayNotification', () => {
+	let registry;
+
+	beforeAll( () => {
+		registry = createTestRegistry();
+	} );
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.test.js:41]: We usually reset the registry after each test case (i.e. in `beforeEach()`). Is there a reason we're doing it here once?

File: assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.test.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		// Set the notification as dismissed.
+		registry
+			.dispatch( CORE_USER )
+			.receiveGetDismissedItems( [
+				RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION,
+			] );
+
+		const { container, waitForRegistry } = render(
+			<ViewContextProvider
+				value={ VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY }
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.test.js:119]: We should change this to be the main dashboard as the component would not render in the view-only dashboard even when it is not dismissed.

```suggestion
				value={ VIEW_CONTEXT_MAIN_DASHBOARD }
```

File: assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss [hunk 4]
```diff
@@ -98,4 +98,18 @@
 			padding-top: 0;
 		}
 	}
+
+	&.googlesitekit--overlay-notification--reader-revenue-manager {
+		max-width: 348px;
```
Comment (nfmohit) [inline assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:103]: This max-width should be applied only on viewport `$bp-tablet` and up.

File: assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss [hunk 5]
```diff
@@ -98,4 +98,18 @@
...
+
+		.mdc-button--raised {
+			@extend .mdc-button--raised;
```
Comment (nfmohit) [inline assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:106]: This looks redundant. Do you think this is necessary?

File: assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss [hunk 6]
```diff
@@ -98,4 +98,18 @@
...
+			font-weight: $fw-medium;
+			padding: 10px $grid-gap-phone;
```
Comment (nfmohit) [inline assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:108]: This won't be needed if the `Button` component is used.

File: assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss [hunk 7]
```diff
@@ -98,4 +98,18 @@
 			padding-top: 0;
 		}
 	}
+
+	&.googlesitekit--overlay-notification--reader-revenue-manager {
```
Comment (nfmohit) [inline assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:102]: I would suggest moving the styles for this component to its own stylesheet, e.g. `assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss`.

File: assets/svg/graphics/reader-revenue-manager-introductory-graphic-desktop.svg
Comment (nfmohit) [inline assets/svg/graphics/reader-revenue-manager-introductory-graphic-desktop.svg:1]: It looks like there is an issue with the exported graphic. There is a shadow around the subject which does not appear in the Figma designs.

File: assets/svg/graphics/reader-revenue-manager-introductory-graphic-mobile.svg
Comment (nfmohit) [inline assets/svg/graphics/reader-revenue-manager-introductory-graphic-mobile.svg:1]: It looks like there is an issue with the exported graphic. There is a shadow around the subject which does not appear in the Figma designs.

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for your nice work on this, @ankitrox . Please take a look at my comments and feel free to reach out if you have any questions, thanks!

File: assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.js
[the remark this answers is not in the document]
Comment (ankitrox) [reply assets/js/modules/reader-revenue-manager/components/PublicationApprovedOverlayNotification.js:1]: I agree, it makes more sense.

File: assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss (hunk 5 above)
Comment (ankitrox) [reply assets/sass/components/overlay-notification/_googlesitekit-overlay-notification.scss:106]: Removed this one.

File: assets/svg/graphics/reader-revenue-manager-introductory-graphic-desktop.svg
Comment (ankitrox) [reply assets/svg/graphics/reader-revenue-manager-introductory-graphic-desktop.svg:1]: I've asked @sigal-teller to kindly look into the issue.

File: assets/svg/graphics/reader-revenue-manager-introductory-graphic-mobile.svg
Comment (ankitrox) [reply assets/svg/graphics/reader-revenue-manager-introductory-graphic-mobile.svg:1]: I've asked @sigal-teller to kindly look into the issue.

**Comment by ankitrox** [comment]
Thanks @nfmohit for your review.

I've addressed the feedback added and sending this your way for another review.

File: assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+		// Dismiss the notification, which also dismisses it from
+		// the current user's profile with the `dismissItem` action.
+		dismissOverlayNotification(
+			RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION
+		);
+	};
+
+	if ( isDismissing ) {
+		return null;
+	}
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js:96]: I feel this is redundant and may even have unexpected side effects, for example, if the dismissal request fails for any reason (e.g. a network error), we may see a flash of the notification disappearing and reappearing. We also do not have this check for other overlay notifications. What do you think?

```suggestion
```

File: assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+			<div className="googlesitekit-overlay-notification__actions">
+				<Button
+					tertiary
+					disabled={ isDismissing }
+					onClick={ dismissNotification }
+				>
+					{ __( 'Maybe later', 'google-site-kit' ) }
+				</Button>
+
+				<Button
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js:129]: This button should also be disabled when `isDismissing` is truthy.

File: assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+					href={ serviceURL }
+					trailingIcon={ <ExternalIcon width={ 14 } height={ 14 } /> }
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js:131]: According to the Figma designs, I believe the dimensions here should be `13`.

```suggestion
					trailingIcon={ <ExternalIcon width={ 13 } height={ 13 } /> }
```

File: assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.test.js [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+ */
+
+/**
+ * Internal dependencies
+ */
+import { createTestRegistry } from '../../../../../../tests/js/utils';
+import { render } from '../../../../../../tests/js/test-utils';
+import PublicationApprovedOverlayNotification, {
+	RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION,
+} from '../dashboard/PublicationApprovedOverlayNotification';
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.test.js:26]:
```suggestion
} from './PublicationApprovedOverlayNotification';
```

File: assets/js/modules/reader-revenue-manager/components/dashboard/index.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+/**
+ * RRM Dashboard components.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/dashboard/index.js:4]:
```suggestion
 * RRM dashboard components.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
```

File: assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+.googlesitekit-overlay-notification {
+	&.googlesitekit-reader-revenue-manager-publication-approved-notification {
```
Comment (nfmohit) [inline assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss:20]: 1. We usually have most of our styles as descendants of the `.googlesitekit-plugin` class.
2. We can reduce the specificity of this selector, i.e. we don't need the `.googlesitekit-overlay-notification` sibling selector.
3. We usually follow a convention where we add an empty line before every selector.

Please see suggestion:

```suggestion
.googlesitekit-plugin {

	.googlesitekit-reader-revenue-manager-publication-approved-notification {

```

File: assets/sass/components/reader-revenue-manager/_index.scss [hunk 14]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
```
Comment (nfmohit) [inline assets/sass/components/reader-revenue-manager/_index.scss:3]: Extreme nit, but while we're at it:

```suggestion
/**
 * Reader Revenue Manager styles.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
```

File: assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+		.mdc-button--raised {
+			font-weight: 500;
+		}
```
Comment (nfmohit) [inline assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss:23]: The button text and icon should have a gap of 6px, and we should use appropriate variable for the font weight:

```suggestion
		.mdc-button--raised {
			gap: 0 6px;
			font-weight: 500;
		}
```

File: tests/backstop/reference/google-site-kit_Modules_ReaderRevenueManager_Components_PublicationApprovedOverlayNotification_0_document_0_small.png
Comment (nfmohit) [inline tests/backstop/reference/google-site-kit_Modules_ReaderRevenueManager_Components_PublicationApprovedOverlayNotification_0_document_0_small.png:1]: While the graphics have been updated, for some reason, the VRT references still have the shadow around the graphic. I think this is why VRT is failing. Could you take a look?

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for addressing my feedback, @ankitrox. I apologise for the back and forth, but I have left another set of feedback on the follow-up, which I hope should be quite straightforward. Please let me know if you have any questions, thank you!

**Comment by ankitrox** [comment]
Thanks for reviewing this @nfmohit 

I've addressed the comments added by you and sending your way for another review. I've also updated the VRT reference images in the PR.

File: assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+.googlesitekit-plugin {
+
+	&.googlesitekit-reader-revenue-manager-publication-approved-notification {
```
Comment (nfmohit) [inline assets/sass/components/reader-revenue-manager/_googlesitekit-rrm-overlay-notification.scss:21]: The specificity is incorrect here, causing the styles to be broken.

```suggestion
	.googlesitekit-reader-revenue-manager-publication-approved-notification {
```

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for addressing my feedback, @ankitrox.

It looks like the CSS caused the styles to break which I've highlighted below. It looks like a small fix, so I'll try and see if that resolves the issue.

**Review by nfmohit — APPROVED** [review]
LGTM ✅

**Pull request merged**